tests/foreman/ui/test_host.py

```diff
@@ -2597,3 +2612,47 @@ def test_positive_remove_role_in_new_ui(self):
         up on the UI
 
     """
+
+
+@pytest.mark.tier2
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_positive_tracer_enable_reload(prepared_host, target_sat):
+    """Using the new Host UI,enable tracer and verify that the page reloads
+
+    :id: c9ebd4a8-6db3-4d0e-92a2-14951c26769b
+
+    :caseComponent: Katello-tracer
+
+    :Team: Phoenix
+
+    :CaseLevel: System
+
+    :Steps:
+        1. Register a RHEL host to Satellite.
+        2. Prepare Katello-tracer to be installed
+        3. Navigate to the Traces tab in New Host UI
+        4. Enable tracer using the popup
+
+    :expectedresults: The Tracer tab message updates accordingly during the process, and displays
+        the state the correct Title
+
+    """
+    host = target_sat.api.Host().search(query={'search': prepared_host.hostname})[0].read_json()
+    with target_sat.ui_session() as session:
+        session.organization.select(host['organization_name'])
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
+        assert title == "Traces are not enabled"
+        session.host_new.enable_tracer(prepared_host.hostname)
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
+        assert title == "Traces are being enabled"
+        # Waiting on traces to be enabled. Not optimal, but not sure how else to do it
+        wait_for(
+            lambda: session.host_new.get_tracer_title(prepared_host.hostname)
+            != "Traces are being enabled",
+            timeout=1800,
+            delay=5,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
```

Comment:
It does return the lambda value, unfortunately that value is a boolean, since I'm doing a comparison. No clean way I can think of to have this function properly and also return the pre comparison value.

Reply:
If you look closely at my suggestion, I have moved the comparison to `fail_condition`. But this is non-blocking and could be eventually addressed in #11112.